conan/tools/env/environment.py

```diff
@@ -486,11 +495,43 @@ def save_sh(self, file_location, generate_deactivate=True):
         content = "\n".join(result)
         save(file_location, content)
 
+    def save_fish(self, file_location, generate_deactivate=True):
+        filepath, filename = os.path.split(file_location)
+        deactivate_file = os.path.join(filepath, "deactivate_{}".format(filename))
+        deactivate = textwrap.dedent("""\
+                   echo "echo Restoring environment" > "{deactivate_file}"
+                   for v in {vars};
+                       set is_defined true
+                       set value (printenv $v) || set is_defined "" || true
+                       if [ -n "$value" ]
+                           or [ -n "$is_defined" ];
+                           echo "set --export $v '$value'" >> "{deactivate_file}";
+                       else; echo set --erase $v >> "{deactivate_file}";
```

Comment:
I am not an expert in `fish` shell, but it appears as a subject for an injection.
if I set `v = something && rm -rf /` it will evaluate into `echo set --erase something && rm -rf /`, which is dangerous.
I don't know for sure, but perhaps it needs some sort of quoting or sanitation to avoid interpreting values as commands.

Reply:
From Fish' help:
> The names given to variables and functions (so called “identifiers”) have to follow certain rules:
 A variable name cannot be empty. It can contain only letters, digits, and underscores. It may begin and end with any of those characters.

It will break sooner than this if any bad char is found, but I guess it does not hurt to be triple-sure :)